Prepare v0.6.0

behackl opened the pull request:
<!--changelog-start-->
<!--changelog-end-->

## Changelog
- Added generated changelog
- Bumped version to v0.6.0

behackl commented:
I would still like to get #1430 and #1431 in. Please everyone @ManimCommunity/core check out the generated changelog (as soon as RTD finished the build)!

AntonBallmaier commented on docs/source/changelog/0.6.0-changelog.rst:
```diff
@@ -0,0 +1,439 @@
+* Abhijith Muthyala
+* Adam Ryczkowski +
+* Alex Lembcke +
+* Anton Ballmaier +
```
```suggestion
* Anton Ballmaier
```

I have authored a patch in 0.5 as well. Other names may be wrong as well?

jsonvillanueva replied:
Can you link the PR, this may help to debug the dev_changelog.py script.

jsonvillanueva replied:
Nvm, I found it: 
![image](https://user-images.githubusercontent.com/29168469/116787067-e2911380-aa56-11eb-997b-fd615f6e54c1.png)

The issue here is that your name changed between v0.4.0 -> v0.5.0 and v0.5.0 -> v0.6.0  from `AntonBallmaier` to `Anton Ballmaier` with a space -- our script works by pulling the author name and comparing author names from the set of past contributors to current contributors. I suppose we could do it based on a combination of email/user

AntonBallmaier commented on docs/source/changelog/0.6.0-changelog.rst:
```diff
@@ -0,0 +1,439 @@
+* `#1122 <https://github.com/ManimCommunity/manim/pull/1122>`__: Support for specifying the interpolation algorithms for individual ImageMobjects
+
+
+* `#1283 <https://github.com/ManimCommunity/manim/pull/1283>`__: Set default value of keyword ``random_seed`` in Scene to None (was 0 and fixed before)
```
```suggestion
* `#1283 <https://github.com/ManimCommunity/manim/pull/1283>`__: Set default value of keyword ``random_seed`` in Scene to ``None`` (was 0 and fixed before)
```

jsonvillanueva replied:
Fixed the title and also added a reference to :class:`~.Scene`

AntonBallmaier commented on docs/source/changelog/0.6.0-changelog.rst:
```diff
@@ -0,0 +1,439 @@
+   - Raises TypeError when a non-Mobject is added
+   - Filters out incoming duplicate submobjects if at least one instance of that submobject exists in the list
+
+* `#1249 <https://github.com/ManimCommunity/manim/pull/1249>`__: Set corners of :class:`~.Rectangle` in counterclockwise direction
+   Set the corners of `Rectangle` in counterclockwise direction to be consistent with other geometry `Mobject`s. `Transform`ations between `Rectangle` and other geometry `Mobject`s look better now.
```
```suggestion
* `#1249 <https://github.com/ManimCommunity/manim/pull/1249>`__: Set corners of :class:`~.Rectangle` in counterclockwise direction to be consistent with other geometry `Mobject`s. `Transform`ations between `Rectangle` and other geometry `Mobject`s look better now.
```

AntonBallmaier replied:
It sound a bit repetitive.

jsonvillanueva replied:
I modified the body of this: https://github.com/ManimCommunity/manim/pull/1249
@abhi-6988 @abhi3019  Please review and modify if you find the language undescriptive.

abhi3019 replied:
> I modified the body of this: #1249

Looks good, no issues with that thanks! I've also updated the title and changelog of #1385, please have a look

abhi3019 commented on docs/source/changelog/0.6.0-changelog.rst:
```diff
@@ -0,0 +1,439 @@
+   The parameter `path_arc` of :class:`~.Transform` now works with the `.animate` syntax
+
+* `#1364 <https://github.com/ManimCommunity/manim/pull/1364>`__: Added :meth:`~.Mobject.match_points`
+   - Added :func:`~.Mobject.become_points`, which transforms the points, positions and submobjects of a Mobject to match that of the other while keeping style unchanged.
```
CC @Darylgolden 
```suggestion
   - Added :meth:`~.Mobject.match_points`, which transforms the points, positions and submobjects of a Mobject to match that of the other while keeping style unchanged.
```

Darylgolden replied:
LGTM

naveen521kk replied:
can you change it in the PR title, then behackl would render this again?

behackl replied:
I've already taken care of it. 👍

behackl commented:
I fixed some more broken formatting and fixed a few links; I'm more or less fine with the current state of this. I'll be away for a few hours, everyone with write access can re-run the script though (or push changes to the changelog directly).

Darylgolden commented:
Also, can I ask how the list of contributors who reviewed PRs is determined? I tried to review several PRs this release, even though I don't have write access.

jsonvillanueva commented:
> Also, can I ask how the list of contributors who reviewed PRs is determined? I tried to review several PRs this release, even though I don't have write access.

https://github.com/ManimCommunity/manim/blob/master/scripts/dev_changelog.py#L92

If you go to the `Files changed` tab there's a green dropdown button that allows you to make Reviews. I don't think comments are included in the GitHub API request.

Darylgolden commented:
> > Also, can I ask how the list of contributors who reviewed PRs is determined? I tried to review several PRs this release, even though I don't have write access.
> 
> https://github.com/ManimCommunity/manim/blob/master/scripts/dev_changelog.py#L92
> 
> If you go to the `Files changed` tab there's a green dropdown button that allows you to make Reviews. I don't think comments are included in the GitHub API request.

I'm pretty sure I did that several times.

jsonvillanueva commented:
> I'm pretty sure I did that several times.

I'm not really sure what the issue is. We're not doing any filtering on whether the review is of type `approves/request changes/comments` and you do appear in at least one REST API query: https://api.github.com/repos/ManimCommunity/manim/pulls/1347/reviews

Darylgolden commented:
I initially thought the issue was write access, but there are people without write access in the reviewers list too 😕

Darylgolden commented:
I also think @hydrobeam reviewed some PRs but isn't included too, such as at #1430

kolibril13 commented:
Thanks for making the pr, I just saw that my name is also not there in the "have been reviewed" section.

behackl commented:
For those not in the reviewer list: can you check whether you have a name set in your github profile? If not, then that's the issue. I think we can fix the situation by using the login name if no actual name is set.

kolibril13 commented:
> For those not in the reviewer list: can you check whether you have a name set in your github profile? If not, then that's the issue. I think we can fix the situation by using the login name if no actual name is set.

I did not set my name, but it is set now :)

hydrobeam commented:
I've also set my name 🚀

naveen521kk commented on docs/source/changelog/0.6.0-changelog.rst:
```diff
@@ -0,0 +1,456 @@
+* Adam Ryczkowski +
+* Alex Lembcke +
+* Anton Ballmaier +
+* Aron
+* Aron Fischer
```
I think both the names are the same right? @cobordism

Darylgolden commented:
I've set my (fake) name

behackl commented:
PRs #1434 (obviously) as well as #1282 #1200 #1259 were not yet included in the list because their merge message have either been modified, or because they were merged manually/differently. I've added them now manually.

Also, I've changed the logic of how authors are determined -- it now also uses the GitHub API, the names in the authors and reviewers list should now be consistent.

jsonvillanueva commented on docs/source/changelog/0.6.0-changelog.rst:
```diff
@@ -0,0 +1,474 @@
+
+
+* `#1259 <https://github.com/ManimCommunity/manim/pull/1259>`__: Removed deprecated CairoText class
+   - Removed `CairoText` class from `manim.mobject.text_mobject`
+    - Removed `'CairoText'` from `__all__`
+    - Removed reference to `CairoText` in docstring
```
This indentation needs to be fixed, but I think this s3ection is honestly pretty redundant:
```suggestion

```

jsonvillanueva commented on docs/source/changelog/0.6.0-changelog.rst:
```diff
@@ -0,0 +1,474 @@
+----------------
+
+* `#1347 <https://github.com/ManimCommunity/manim/pull/1347>`__: Restructure vector_field module and add documentation
+   `VectorField` is renamed to `ArrowVectorField` and a new class `VectorField` is added as a superclass for `ArrowVectorField` and `StreamLines`. Also the class `AnimatedStreamLines` is removed. It's functionality is moved to `StreamLines`. Added a lot of new options when working with vector fields. `ShowPassingFlashWithThinningStrokeWidth` was moved to the indication module.
```
```suggestion
   :class`~.VectorField` is renamed to :class:`~.ArrowVectorField` and a new class :class:`~.VectorField` is added as a superclass for :class:`~.ArrowVectorField` and :class:`~.StreamLines`. `AnimatedStreamLines` is removed. It's functionality is moved to :class:`~.StreamLines`. Added a lot of new options when working with vector fields. :class:`~.ShowPassingFlashWithThinningStrokeWidth` was moved to the indication module.
```

jsonvillanueva reviewed (approved):
I'm approving this in advance with these changes (Gotta make the first PR look great and remove the malformatted bullets); however, I'd like to mention that there could be a more consistent grammar/authoring of this text. As is, it appears to be written by various authors and in some cases there's redundancies between the PR title/body text. We'd be here all day doing nitpicks and the meaning is understood nonetheless.

behackl commented:
I won't regenerate the changelog at this point, I think it is good to go. (Certainly not written uniformly, and some information might be redundant -- but I don't think there is a wide margin for improvement here.)

Pretty impressive work everyone, and thanks for reviewing the changelog! I'll merge and then release soon™️, if you have further comments or remarks, it's either now or never.
